api/document/json_importer/annotations.py

```diff
@@ -0,0 +1,94 @@
+from collections import defaultdict
+from typing import Callable, DefaultDict, Dict, Iterator, List, Tuple, Type
+
+from document.models import Annotation, ExternalLink, FootnoteCitation
+from document.tree import JSONAwareCursor, PrimitiveDict
+
+Annotator = Callable[
+    [JSONAwareCursor, PrimitiveDict, int],
+    Annotation
+]
+
+annotators: Dict[str, Annotator] = {}
+
+
+def annotator(fn: Annotator):
+    annotators[fn.__name__] = fn
+    return fn
+
+
+@annotator
+def footnote_citation(cursor: JSONAwareCursor, content: PrimitiveDict,
```

Comment:
Should this and the other annotation deserializing logic be in the corresponding serializer (e.g. `FootnoteCitationSerializer`)? That seems like the DRF way.

Reply:
Maybe? The thing I am constantly finding confusing about all this is that DRF really seems to want the validation process to return a plain old python dictionary with primitive values, not having anything to do with models, which is why this PR has the deserialization deal _only_ with validation/normalization of input and the JSON importer deal _only_ with actually turning that validated structure into a thing the database can understand.

That said, I think my fear of `DocCursor` is impacting the way I'm working on things in a more general way.  It's also part of why I found it easier to go through the XML intermediary during the PDF import, in #767.  Even without DRF's nudging, I'm finding that I tend to veer towards intermediary abstractions that process data to _prepare_ it for manipulation via a `DocCursor`, and then do the `DocCursor` stuff in a separate step.

Anyways, I guess maybe I should focus a bit on actually learning `DocCursor` better, perhaps by writing some documentation for it, as I had brought up in #767. If I feel more confident using that, I'll probably be more likely to submit PRs that have less unnecessary abstractions in them, not to mention making me better at reviewing code that uses the class!